#!/usr/bin/env python
PACKAGE="image_manip"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("frame_rate", double_t, 1, "frame rate Hz", 20, 0.1, 50)
gen.add("capture_single", bool_t, 1, "capture single image", False)
gen.add("index", int_t, 4, "current index", 0, 0, 512)
gen.add("start_index", int_t, 8, "start index", 0, 0, 512)
gen.add("use_live_frame", bool_t, 8, "use live frame as final frame of animation", True)
gen.add("capture_continuous", bool_t, 8, "save all received images", False)
gen.add("restrict_size", bool_t, 8, "limit deque size", False)
gen.add("max_size", int_t, 8, "limit deque size", 10, 1, 512)

exit(gen.generate(PACKAGE, "image_deque", "ImageDeque"))
